tests/test_unittest/test_gtc/common_utils.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+#
+# GTC Toolchain - GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2021, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from typing import Any, Callable, Set
+
+import factory
+
+from eve import concepts, traits, type_definitions, visitors
+from gtc import common
+
+
+def undefined_symbol_list(
+    symbol_creator: Callable[[str], Any], *fields_to_collect: str, list_creator=None
+) -> factory.LazyAttribute:
+    class CollectSymbolsAndRefs(visitors.NodeVisitor):
+        def visit_Node(self, node: concepts.Node, *, symbols: Set[str], refs: Set[str]) -> None:
+            for name, metadata in node.__node_children__.items():
+                type_ = metadata["definition"].type_
+                if isinstance(type_, type):
+                    if issubclass(type_, type_definitions.SymbolName):
+                        symbols.add(getattr(node, name))
+                    elif issubclass(type_, type_definitions.SymbolRef):
+                        refs.add(getattr(node, name))
```

Comment:
Is the `isintance(type_, type)` there because `type_` can be something that is not a `type`? If so, is that only the case for `None` (in which case we could change to `if type_ is not None`) or is it also allowed to be a string representing a type (in which case that should be handled as well)?

Reply:
Good question, I just copied the relevant part from https://github.com/GridTools/gt4py/blob/394cafb03cbd01feb276e8a9ed872770557c9f32/src/eve/traits.py#L35, but I don’t know if there is any relevant case where `type_` is not a `type` to be honest.

Reply:
I checked: `type_` is often an instance of one of the `typing.*` type aliases, so the check is actually correct and required.

Reply:
👍